### ISSUE TITLE
Replacing <CI Provider> with GitHub Actions

### ISSUE THREAD

**Issue by JasonEtco** [JasonEtco/jasonet.co#22]
I've been using Actions more and more in my every-day projects, to the point that I'm very unlikely to ever use Travis or Circle CI for a new project. It's just easier to create a `main.workflow` file and call it a day. I'd like to write a post sharing what I think is the most common, and also most optimized, workflow for CI (for Node projects, with hints to other things).

h/t @gr2m for the suggestion ✨

**Comment by JasonEtco** [comment]
Also, a note - I'll probably start off with something like "Actions is more than just CI - but its also CI"

**Comment by JasonEtco** [comment]
Also spoiler:

```workflow
workflow "Test my code" {
  on = "push"
  resolves = ["codecov"]
}

action "npm ci" {
  uses = "docker://node:alpine"
  runs = "npm"
  args = "ci"
}

action "npm test" {
  needs = "npm ci"
  uses = "docker://node:alpine"
  runs = "npm"
  args = "test"
}

action "codecov" {
  needs = "npm test"
  uses = "docker://node"
  runs = "npx"
  args = "codecov"
  secrets = ["CODECOV_TOKEN"]
}
```

**Comment by gr2m** [comment]
In the post, will you list common CIs such as Circle CI and Travis CI and explain how to get from their respective configuration files to implement the same using workflows? If not, I'd maybe call the post "GitHub Actions for Continuous Integration"

**Comment by JasonEtco** [comment]
> will you list common CIs such as Circle CI and Travis CI and explain how to get from their respective configuration files to implement the same using workflows?

Yes! I'll go into that too. Circle in particular maps nicely because of how it has Docker images for environments; not sure if Travis does the same thing, but either way it'll work out. Definitely going to mention the "here's how to test in different environments/Node versions" bit, since Travis does that nicely but it's verbose in Actions.

**Comment by JasonEtco** [comment]
Closed by #25
